test/http_test_client.rb

```diff
@@ -5,11 +5,16 @@ class HTTPTestClient
     def code; '200'; end
   end
 
+  class << self
+    attr_accessor :last_url
+  end
+
   attr_accessor :body
 
   def initialize(_config); end
 
-  def get(_url)
+  def get(url)
+    self.class.last_url = url
```

Comment:
I don't think this is a thread-safe implementation. If you want to get rid of the `send` methods in the tests I might recommend a spy that you can later assert on, some kind of `UrlAssertHTTPTestClient` that would expose the information you want, or possibly extracting the URL construction into a separate object that can be tested in isolation.

Overall, I don't know that it really matters though. There is no behavioral change in this PR. The `send` method is certainly not a good practice, but it's contained within the test for its class only - so I don't particularly see it as a big problem. Under normal circumstances we might expose this method publicly, but I don't want to make this part of the public API.

Reply:
It's rewrited with mocks! What do you think now?